File: freya/CustomerService/CustomerService.fs
namespace Customers
open System  
type CustomerService () = 
    let trim (x: string) = x.Trim()
    let splitIntoColumns (line:string) =
        line.Split([|','|]) |> Seq.map trim |> Seq.toList
    let mapToCustomer (columns: string list)=
         {Customer.Empty with 
            AccountNumber = columns |> List.head |> Int32.Parse ; 
            FirstName = List.nth columns 1; 
            LastName = List.nth columns 2;
         }

    let cs = """1,Oskar,Gewalli,
            2,Greta,Skogsberg,
            3,Matthias,Wallisson,
            4,Ada,Lundborg,
            5,Daniel,Örnvik,
            6,Johan,Irisson,
            7,Edda,Tyvinge""".Split(separator =[|'\n'; '\r'|], options=System.StringSplitOptions.RemoveEmptyEntries)
            |> Seq.map (splitIntoColumns) 
            |> Seq.map (mapToCustomer)
        
    interface ICustomerService with
        member x.GetAllCustomers () =
            CustomerOutput.Multiple( cs |> Seq.toList )

        member x.SaveCustomers (cs: CustomerInput)=
            CustomerOutput.Success(false)

File: clean-aspnet-core/CustomerService/CustomerService.cs
using System;
using System.Linq;

namespace Customers
{
    public class CustomerService : ICustomerService
    {
        public ArrayOfCustomer GetAllCustomers()
        {
            return new ArrayOfCustomer
            {
                Customer = new[] 
                    {
                        new Customer { AccountNumber=1, FirstName = "Oskar", LastName = "Gewalli" },
                        new Customer { AccountNumber=2,FirstName = "Greta", LastName = "Skogsberg" },
                        new Customer { AccountNumber=3,FirstName = "Matthias", LastName = "Wallisson" },
                        new Customer { AccountNumber=4,FirstName = "Ada", LastName = "Lundborg" },
                        new Customer { AccountNumber=5,FirstName = "Daniel", LastName = "Örnvik" },
                        new Customer { AccountNumber=6,FirstName = "Johan", LastName = "Irisson" },
                        new Customer { AccountNumber=7,FirstName = "Edda", LastName = "Tyvinge" },
                    }
            };
        }

        public Customer GetCustomerByAccountNumber(int accountNumber)
        {
            return GetAllCustomers().Customer
                .SingleOrDefault(c => c.AccountNumber == accountNumber);
        }

        public ArrayOfCustomer GetCustomers(string lastName)
        {
            return new ArrayOfCustomer
            { 
                Customer= GetAllCustomers().Customer
                    .Where(c => c.LastName == lastName).ToArray()
            };
        }

        public bool SaveCustomer(Customer editedCustomer)
        {
            return false;
        }

        public bool SaveCustomerLastName(string accountNumber, string newName)
        {
            return false;
        }
    }
}
